includes/utils/class-amp-image-dimension-extractor.php

```diff
@@ -150,6 +151,89 @@ private static function register_callbacks() {
 		do_action( 'amp_extract_image_dimensions_batch_callbacks_registered' );
 	}
 
+	/**
+	 * Extract dimensions from filename if dimension exists or from file system.
+	 *
+	 * @param array $dimensions Image urls mapped to dimensions.
+	 *
+	 * @return array Dimensions mapped to image urls, or false if they could not be retrieved
+	 */
+	public static function extract_by_filename_or_filesystem( $dimensions ) {
```

Comment:
This should have a `dataSet` to test with `wp_using_ext_object_cache()` being both `true` and `false`.

Reply:
Sorry, I meant to leave this comment on the test.